src/screens/SettingsScreen.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * @flow
+ */
+
+import type {
+  NavigationScreenConfig,
+  NavigationScreenProp,
+  NavigationState,
+} from 'react-navigation';
+
+import type { GameStoreProps } from 'src/undux/GameStore';
+
+import React from 'react';
+
+import {
+  Image,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+  TextInput,
+  Picker,
+  Button,
+  Alert,
+} from 'react-native';
+
+import { StackNavigator, NavigationActions } from 'react-navigation';
+
+import { withStore } from 'src/undux/GameStore';
+
+import { JpDisplayStyle } from 'src/jp/Types';
+
+type Props = {
+  ...GameStoreProps,
+  navigation: NavigationScreenProp<NavigationState>,
+};
+
+type State = {
+  // nothing
+};
+
+class SettingsScreen extends React.Component<Props, State> {
```

Comment:
you can remove `State` from `React.Component<Props, State>` if there's no state

moreover, since there's no state, you can refactor this as a functional component, which are simpler and have better performance. see the following docs for examples:

https://reactjs.org/docs/components-and-props.html
https://flow.org/en/docs/react/components/#toc-stateless-functional-components

(it's also a cool way to think about components, as functions!)